AWSPinpoint/AWSPinpointEndpointProfile.h

```diff
@@ -99,7 +99,7 @@ NS_ASSUME_NONNULL_BEGIN
  An AWSPinpointEndpointProfile object with the specified context
  @returns AWSPinpointEndpointProfile
  */
-- (instancetype)initWithContext:(AWSPinpointContext *) context;
+- (instancetype)initWithContext:(AWSPinpointContext *) context isRegisteredForRemoteNotifications:(BOOL) isRegisteredForRemoteNotifications;
```

Comment:
This is a public header right?

Reply:
Yeah,  but `AWSPinpointContext` is defined in the `[Internal](https://github.com/aws-amplify/aws-sdk-ios/tree/main/AWSPinpoint/Internal)` folder and those things can't be consumed, right?

At least on my sample app, this init never appeared as an option, and attempting to use `AWSPinpointContext` was greyed out because "it was not imported".

But maybe that's a Swift-only thing? If you prefer I can keep and deprecate the old method, and create a new one with the extra parameter that we use.